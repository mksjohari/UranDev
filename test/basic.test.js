import React from "react";
import ReactDOM from "react-dom";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
const renderer = require("react-test-renderer");
configure({ adapter: new Adapter() });

import Basic from "./basic";
import SignUp from "../src/shared/signUpForm";
// import App from "../src/App";

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    ReactDOM.unmountComponentAtNode(container);
    container.remove();
    container = null;
});

test("example component renders without crashing", () => {
    shallow(<Basic />);
});

test("component matches previous snapshots", () => {
    expect(renderer.create(<Basic />)).toMatchSnapshot();
});

test("example component renders correctly", () => {
    ReactDOM.render(<Basic />, container);
    expect(document.querySelector("h1").innerHTML).toBe("Basic Test");
});

// const wrapper = mount(<Basic />);
// expect(wrapper.state("error")).toEqual(null);

// const wrapper = shallow(<SignUp />).dive();

// jest.mock('react-router-dom', () => ({
//   useLocation: jest.fn().mockReturnValue({
//     pathname: '/another-route',
//     search: '',
//     hash: '',
//     state: null,
//     key: '5nvxpbdafa',
//   }),
// }));
