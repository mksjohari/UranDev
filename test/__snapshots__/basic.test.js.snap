// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches previous snapshots 1`] = `
<div>
  <h1>
    Basic Test
  </h1>
  <p>
     This is a basic Test Component
  </p>
</div>
`;
